streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -118,6 +120,10 @@ public void init(final InternalProcessorContext context) {
         } catch (final Exception e) {
             throw new StreamsException(String.format("failed to initialize processor %s", name), e);
         }
+
+        // revived tasks could re-initialize the topology,
+        // in which case we should reset the flag
+        closed = false;
```

Comment:
Should we assert that it's in `closed` when `init` is called?

Reply:
SG.